### DISCUSSION TITLE
Where MMS are stored ?

### DISCUSSION THREAD

**Discussion by s0rau** [tmo1/sms-ie#147] [General]
I'm searching where MMS are stored ?
SMS are stored here : /data/data/com.android.providers.telephony/  but  I don't find any MMS picture

**Comment by tmo1** [comment]
AFAIK, the specific location of MMS binary data in the filesystem is not part of the official Android APIs. SMS I/E does not use this location - all its MMS access is via the ContentResolver APIs. That being said, on my devices, the location is apparently `/data/user_de/0/com.android.providers.telephony/app_parts/`. (This is the location reported in `Telephony.Mms.Part#_DATA`, taken from SMS I/E exports; I have not inspected the filesystem myself.)
